src/pages/settings/Profile/ProfilePage.js

```diff
@@ -241,17 +250,36 @@ class ProfilePage extends Component {
                         <Text style={[styles.mt6, styles.mb6]}>
                             {this.props.translate('profilePage.tellUsAboutYourself')}
                         </Text>
-                        <FullNameInputRow
-                            firstName={this.state.firstName}
-                            firstNameError={PersonalDetails.getMaxCharacterError(this.state.hasFirstNameError)}
-                            lastName={this.state.lastName}
-                            lastNameError={PersonalDetails.getMaxCharacterError(this.state.hasLastNameError)}
-                            onChangeFirstName={firstName => this.setState({firstName})}
-                            onChangeLastName={lastName => this.setState({lastName})}
-                            style={[styles.mt4, styles.mb4]}
-                        />
+
+                        <View style={[styles.flexRow, styles.mt4, styles.mb4]}>
+                            <View style={styles.flex1}>
+                                <TextInput
+                                    inputID="firstName"
+                                    name="fname"
+                                    label={this.props.translate('common.firstName')}
+                                    value={this.state.firstName}
+                                    hasError={this.state.hasFirstNameError}
+                                    errorText={PersonalDetails.getMaxCharacterError(this.state.hasFirstNameError)}
+                                    onChangeText={firstName => this.setState({firstName})}
+                                    placeholder={this.props.translate('profilePage.john')}
+                                />
+                            </View>
+                            <View style={[styles.flex1, styles.ml2]}>
+                                <TextInput
+                                    inputID="lastName"
+                                    name="lname"
+                                    label={this.props.translate('common.lastName')}
+                                    value={this.state.lastName}
+                                    hasError={this.state.hasLastNameError}
+                                    errorText={PersonalDetails.getMaxCharacterError(this.state.hasLastNameError)}
+                                    onChangeText={lastName => this.setState({lastName})}
+                                    placeholder={this.props.translate('profilePage.doe')}
+                                />
+                            </View>
+                        </View>
                         <View style={styles.mb6}>
                             <Picker
+                                inputID="pronounsPicker"
```

Comment:
this one too

Reply:
Looks like there is a bug with the Picker control. If the `inputID` is removed from the component, changing of values works fine. However, as soon as I add the `inputID`, it stops working. 

See the attached screencast.

https://user-images.githubusercontent.com/911879/175491513-9ffb4761-bc75-44f2-96d5-58ad3bc5f1ff.mp4

Is it a known bug or am I doing something wrong?

Reply:
It looks like you are still providing other props to Picker, e.g. onInputChange and value. onInputChange is automatically provided by Form when inputID is passed so the state value never changes which causes the behavior you are seeing. When using these inputs with Form, you shouldn't pass onInputChange handlers.

Reply:
@luacmartins We update a state variable inside `onInputChange` which is used at multiple places.
```
onInputChange={(pronouns) => {
    const hasSelfSelectedPronouns = pronouns === CONST.PRONOUNS.SELF_SELECT;
    this.setState({
        pronouns: hasSelfSelectedPronouns ? '' : pronouns,
        hasSelfSelectedPronouns,
    });
}}
```

And `value` is required for passing the default value to the picker. 

Without using these two props, we won't be able to (i) update the state variable (ii) show defaultValue for the picker. 

`Picker` doesn't seem to be have used with `Form` throughout the project. It is however used with `FormScrollView` and has both `onInputChange` and `value`.

What do you advise?

Reply:
>We update a state variable inside onInputChange which is used at multiple places.

The idea of using form is that we can get rid of local state.

>And value is required for passing the default value to the picker.

Value should NOT be passed when using inputs inside form. The default value is passed to the input automatically.

>Without using these two props, we won't be able to (i) update the state variable (ii) show defaultValue for the picker.

The Form component takes care of update the value and displaying the correct one.

>Picker doesn't seem to be have used with Form throughout the project.

We are in the process of refactoring all our forms to use the new Form component, so that's why you don't see much usage yet.

>What do you advise?

For now, you can connect ProfilePage to the form draft Onyx key and use the draft value stored to show/hide the self-select pronoun input.